src/mmw/js/src/user/controllers.js

```diff
@@ -7,6 +7,7 @@ var App = require('../app'),
 var SignUpController = {
     signUp: function() {
         new views.SignUpModalView({
+            app: App,
```

Comment:
That's correct. `app.js` imports `user/views.js`so it can instantiate a `LoginModalView`: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/app.js#L58. Also, because of their inherent nature, all user views affect or manipulate the `App.user` state, thus need access to `App`. We pass it in as a dependency to avoid circular references.

Since `app.js` only uses `user/views.js` for one specific operation, if we can refactor that away then we can simply require `App` in `user/views.js`. One such way would be to make a `/login` route which opens the `LoginModalView`, and the `app.js` simply triggers that view. But that can be complicated if one is trying to login from the `/model` page and already has unsaved work in progress that they'd like to login to save: changing the URL at that point seems hairy.

If we can think of another way of removing this dependency, that would be great.